algebras/testkit/src/main/scala/endpoints4s/algebra/server/BasicAuthenticationTestSuite.scala

```diff
@@ -62,6 +62,23 @@ trait BasicAuthenticationTestSuite[T <: BasicAuthenticationTestApi] extends Endp
       }
     }
 
+    "fall through to another endpoint if url doesn't match" in {
```

Comment:
In the case you provided, the URL _does_ match, I believe the right behavior is to reject the request with an `Unauthorized` response.

Reply:
@julienrf does it make sense?

Reply:
Aah OK, I see now that there are several endpoints starting with `/users`: `protectedEndpoint`, `unprotectedEndpoint`, and `protectedEndpointWithParameter`. I thought you were using both `protectedEndpoint` and `unprotectedEndpoint` in the tests, but now I see that you use `unprotectedEndpoint` and `protectedEndpointWithParameter`.

In that case, indeed, the URL `/users` does not match the `protectedEndpointWithParameter` but still matches the `unprotectedEndpoint`, so the response should indeed be `"Unprotected"`! :+1: